esp-hal/src/lib.rs

```diff
@@ -256,8 +257,15 @@ pub fn get_core() -> Cpu {
 ///
 /// Safety: This method should never return UNUSED_THREAD_ID_VALUE
 #[cfg(riscv)]
+#[inline(always)]
 fn get_raw_core() -> usize {
-    riscv::register::mhartid::read()
+    #[cfg(multi_core)]
```

Comment:
Is there a reason for this change?

Reply:
It's just to avoid the CSR-read if we already know it will always be 0. Probably doesn't make a huge difference. We can revert it if it's confusing

Reply:
> 3.1.5. Hart ID Register mhartid
The mhartid CSR is an MXLEN-bit read-only register containing the integer ID of the hardware
thread running the code. This register must be readable in any implementation. Hart IDs might not
necessarily be numbered contiguously in a multiprocessor system, but at least one hart must have a
hart ID of zero. Hart IDs must be unique within the execution environment.

Ah nice, so for single-core chips it will always be zero. Let's keep this as it then!